profiler/src/ProfilerEngine/Datadog.Profiler.Native/Sample.h

```diff
@@ -65,10 +60,29 @@ class Sample
     }
 
     // helpers for well known mandatory labels
-    void SetPid(const std::string& pid);
-    void SetAppDomainName(const std::string& name);
-    void SetThreadId(const std::string& tid);
-    void SetThreadName(const std::string& name);
+    template <typename T>
```

Comment:
Why would we need template here?
Shouldn't it be string&?

Reply:
Here `template <typename T>` + `T&&` means universal reference.
Which means that we can call those function and pass any type of reference (r-value, l-value) and we leave the compiler decide what to do:
- R-Value reference case: the object will be move
- L-Value reference case: just pass by ref

This will help us in prevented useless copies (in the r-value ref case)